react-app/src/components/property/PropertyDialog.tsx

```diff
@@ -0,0 +1,337 @@
+      BuildingPredominateUseId: initialValues?.BuildingPredominateUseId,
+      BuildingConstructionTypeId: initialValues?.BuildingConstructionTypeId,
+      TotalArea: String(initialValues?.TotalArea ?? ''),
+      RentableArea: String(initialValues?.RentableArea ?? ''),
+      BuildingTenancy: initialValues?.BuildingTenancy,
+      BuildingTenancyUpdatedOn: dayjs(initialValues?.BuildingTenancyUpdatedOn),
+    });
+  }, [initialValues]);
+
+  return (
+    <ConfirmDialog
+      title={'Edit building information'}
+      open={open}
+      onConfirm={async () => {
+        const formValues: any = { ...infoFormMethods.getValues(), Id: initialValues.Id };
+        formValues.PID = parseIntOrNull(formValues.PID);
+        formValues.PIN = parseIntOrNull(formValues.PIN);
+        formValues.TotalArea = parseFloatOrNull(formValues.TotalArea);
+        formValues.RentableArea = parseFloatOrNull(formValues.RentableArea);
+        formValues.BuildingTenancyUpdatedOn = formValues.BuildingTenancyUpdatedOn.toDate();
+        api.buildings.updateBuildingById(initialValues.Id, formValues).then(() => postSubmit());
+      }}
+      onCancel={async () => onCancel()}
+    >
+      <FormProvider {...infoFormMethods}>
+        <Box display={'flex'} flexDirection={'column'} gap={'1rem'}>
+          <GeneralInformationForm
+            propertyType={'Building'}
+            adminAreas={
+              adminAreasData?.map((admin) => ({ label: admin.Name, value: admin.Id })) ?? []
+            }
+          />
+          <BuildingInformationForm
+            classificationOptions={classificationData}
+            constructionOptions={constructionTypeData}
+            predominateUseOptions={predominateUseData}
+          />
+        </Box>
+      </FormProvider>
+    </ConfirmDialog>
+  );
+};
+
+interface IPropertyAssessedValueEditDialog {
+  initialValues: Parcel | Building;
+  initialRelatedBuildings: Building[];
+  open: boolean;
+  onCancel: () => void;
+  postSubmit: () => void;
+  propertyType: PropertyType;
+}
+
+export const PropertyAssessedValueEditDialog = (props: IPropertyAssessedValueEditDialog) => {
+  const { initialValues, initialRelatedBuildings, open, onCancel, propertyType, postSubmit } =
+    props;
+  const api = usePimsApi();
+  const assessedFormMethods = useForm({
+    defaultValues: {
+      Evaluations: [],
+      RelatedBuildings: [],
+    },
+  });
+  useEffect(() => {
+    assessedFormMethods.reset({
+      Evaluations: initialValues?.Evaluations?.map((evalu) => ({
+        ...evalu,
+        Value: evalu.Value.replace(/[$,]/g, ''), // TODO: Consider some more robust handling for this at the TypeORM level.
+      })),
+      RelatedBuildings: initialRelatedBuildings?.map((building) => ({
+        Id: building.Id,
+        Evaluations: building.Evaluations?.map((evalu) => ({
+          ...evalu,
+          Value: evalu.Value.replace(/[$,]/g, ''), // Obviously this double map is pretty evil so suggestions welcome.
```

Comment:
Yeah, it comes back as a string from TypeORM because the column type is "money", which is more appropriate than using a regular float type, though storing money as an integer is also a valid option (this is how banks typically do it I believe). I think this is acceptable personally, the values are handled as strings regardless in TextFormField.